packages/wonder-blocks-modal/src/components/__tests__/modal-backdrop.test.js

```diff
@@ -47,19 +49,33 @@ describe("ModalBackdrop", () => {
     });
 
     test("Clicking the backdrop triggers `onCloseModal`", () => {
+        // Arrange
         const onCloseModal = jest.fn();
 
-        // We use `mount` instead of `shallow` here, because the component's
-        // click handler expects actual DOM events.
-        const wrapper = mount(
-            <ModalBackdrop onCloseModal={onCloseModal}>
+        render(
+            <ModalBackdrop
+                onCloseModal={onCloseModal}
+                testId="modal-backdrop-test-id"
+            >
                 {exampleModal}
             </ModalBackdrop>,
         );
 
-        expect(onCloseModal).not.toHaveBeenCalled();
+        const backdrop = screen.getByTestId("modal-backdrop-test-id");
+
+        //Act
 
-        wrapper.simulate("click");
+        // Simulate click
+        //
+        // The component looks for mouseup and mousedown instead of just click
+        // on the backdrop to make sure the user didn't drag into or out from
+        // the modal panel.
+        // eslint-disable-next-line testing-library/prefer-user-event
+        fireEvent.mouseDown(backdrop);
+        // eslint-disable-next-line testing-library/prefer-user-event
+        fireEvent.mouseUp(backdrop);
```

Comment:
`userEvent.click()` should be simulating the mousedown and mouseup events as well.  Using `fireEvent` should only be necessary when the mousedown is on one element and the mouseup is on different one.